docs/nx-cloud/reference/launch-templates.md

```diff
@@ -1,42 +1,353 @@
+      - name: Install Rust
+        script: |
+          curl --proto '=https' --tlsv1.3 https://sh.rustup.rs -sSf | sh -s -- -y
+          source "$HOME/.cargo/env"
+          rustup toolchain install 1.70.0
+          # persist cargo bin into PATH
+          echo "PATH=$HOME/.cargo/bin:$PATH" >> $NX_CLOUD_ENV
+```
+
+These templates can be used by passing the number of agents desired, and the template name via `--distribute-on` when starting your CI run.
+
+```
+nx-cloud start-ci-run --distribute-on="4 my-linux-medium-js"
+```
+
+```
+nx-cloud start-ci-run --distribute-on="4 my-linux-large-js"
+```
+
+```
+nx-cloud start-ci-run --distribute-on="4 my-linux-large-rust"
+```
+
+## Reusable Steps
+
+You can find the [list of Nx Cloud reusable steps here](https://github.com/nrwl/nx-cloud-workflows/tree/main/workflow-steps).
+
+## Pass Environment Variables to Agents
+
+If you need to send environment variables to agents, you can use the [--with-env-vars](/ci/reference/nx-cloud-cli#withenvvars) flag on the `nx-cloud start-ci-run` command. You can pass a specific list of environment variables like this:
+
+```
+nx-cloud start-ci-run --distribute-on="8 linux-medium-js" --with-env-vars="VAR1,VAR2"
+```
+
+Or pass all the environment variables except OS-specific ones with this `--with-env-vars="auto"`:
+
+```
+nx-cloud start-ci-run --distribute-on="8 linux-medium-js" --with-env-vars="auto"
+```
+
+## Pass Values Between Steps
+
+If you need to pass a value from one step to another step such as assigning the value to an existing or new environment variable. You can write to the `NX_CLOUD_ENV` environment file.
+
+Commonly this is used for redefining the `PATH` or setting options for various toolchains.
+
+```yaml {% fileName="./nx/workflows/agents.yaml" %}
+launch-templates:
+  my-template-name:
+    init-steps:
+      - name: Set PATH
+        script: echo "PATH=$HOME/.cargo/bin:$PATH" >> $NX_CLOUD_ENV
+      - name: Check PATH
+        script: |
+          # now contains $HOME/.cargo/bin
+          echo $PATH 
+          # can invoke cargo directly because it's in the PATH now. 
+          cargo --version
+```
+
+## Private NPM Registry
+
+If your project consumes packages from a private registry, you'll have to set up an authentication step in a custom launch template. We recommend creating an `.npmrc` file. You can pass the auth token from your main agent, so it's available to the agent machines.
+
+```yaml {% fileName=".nx/workflows/agents.yaml" %}
+launch-templates:
+  my-linux-medium-js:
+    resourceClass: 'docker_linux_amd64/medium'
+    image: 'ubuntu22.04-node20.11-v7'
+    init-steps:
+      - name: Checkout
+        uses: 'nrwl/nx-cloud-workflows/v3.6/workflow-steps/checkout/main.yaml'
+      - name: Auth to Registry
+        script: |
+          echo "@myorg:registry=https://my-private-registry.com/myorg" >> .npmrc
+          echo "//my-private-registry.com/:_authToken=MY_REGISTRY_TOKEN" >> .npmrc
```

Comment:
it would be cool if we could create a custom step to handle this 🤔

Reply:
I agree, though might be a bit more verbose to define inputs for a step than just a few script lines. 

Was thinking if we could take the .npmrc or .yarnrc file if present on the main agent and sync it to the agents so people only configure it via the main agent and wouldn't even need to use a custom launch template. but that edges on the line of "too much magic" for me IMO. 

At min I think a documentation is at min a good step forward (I still need to confirm this works as expected)